lib/client-api/src/args.ts

```diff
@@ -12,24 +15,40 @@ const map = (arg: any, type: ValueType): any => {
     case 'boolean':
       return arg === 'true';
     case 'array':
-      if (!type.value || !Array.isArray(arg)) return undefined;
+      if (!type.value || !Array.isArray(arg)) return INCOMPATIBLE;
       return arg.reduce((acc, item) => {
         const mapped = map(item, type.value as ValueType);
-        return mapped === undefined ? acc : acc.concat([mapped]);
+        return mapped === INCOMPATIBLE ? acc : acc.concat([mapped]);
       }, []);
     case 'object':
-      if (!type.value || typeof arg !== 'object') return undefined;
+      if (!type.value || typeof arg !== 'object') return INCOMPATIBLE;
       return Object.entries(arg).reduce((acc, [key, val]) => {
         const mapped = map(val, (type.value as ObjectValueType)[key]);
-        return mapped === undefined ? acc : Object.assign(acc, { [key]: mapped });
+        return mapped === INCOMPATIBLE ? acc : Object.assign(acc, { [key]: mapped });
       }, {} as Args);
     default:
-      return undefined;
+      return INCOMPATIBLE;
   }
 };
 
 export const mapArgsToTypes = (args: Args, argTypes: ArgTypes): Args => {
   return Object.entries(args).reduce((acc, [key, value]) => {
-    return Object.assign(acc, { [key]: map(value, argTypes[key]?.type) });
+    const mapped = map(value, argTypes[key]?.type);
+    return mapped === INCOMPATIBLE ? acc : Object.assign(acc, { [key]: mapped });
   }, {});
 };
+
+export const combineArgs = (value: any, update: any): Args => {
+  if (isPlainObject(value) && isPlainObject(update)) {
```

Comment:
I agree, that's my preference too. In this case I anticipated we would have another conditional block to deal with arrays, but that turned out not to be necessary. At least not until we start merging arrays rather than overwriting them, which might happen when we start to use sparse arrays in the URL (which the next version of `qs` should support).